runtime/queries/column_time_range.go

```diff
@@ -127,50 +127,72 @@ func handleDuckDBInterval(interval any) (*runtimev1.TimeRangeSummary_Interval, e
 }
 
 func (q *ColumnTimeRange) resolveDruid(ctx context.Context, olap drivers.OLAPStore, priority int) error {
-	rangeSQL := fmt.Sprintf(
-		"SELECT min(%[1]s) as \"min\", max(%[1]s) as \"max\", timestampdiff(SECOND, min(%[1]s), max(%[1]s)) as \"interval\" FROM %[2]s",
+	minSQL := fmt.Sprintf(
+		"SELECT min(%[1]s) as \"min\" FROM %[2]s",
+		safeName(q.ColumnName),
+		safeName(q.TableName),
+	)
+	maxSQL := fmt.Sprintf(
+		"SELECT max(%[1]s) as \"max\" FROM %[2]s",
 		safeName(q.ColumnName),
 		safeName(q.TableName),
 	)
 
-	rows, err := olap.Execute(ctx, &drivers.Statement{
-		Query:            rangeSQL,
+	minRows, err := olap.Execute(ctx, &drivers.Statement{
+		Query:            minSQL,
 		Priority:         priority,
 		ExecutionTimeout: defaultExecutionTimeout,
 	})
 	if err != nil {
 		return err
 	}
-	defer rows.Close()
+	defer minRows.Close()
 
-	if rows.Next() {
-		summary := &runtimev1.TimeRangeSummary{}
-		rowMap := make(map[string]any)
-		err = rows.MapScan(rowMap)
+	maxRows, err := olap.Execute(ctx, &drivers.Statement{
+		Query:            maxSQL,
+		Priority:         priority,
+		ExecutionTimeout: defaultExecutionTimeout,
+	})
+	if err != nil {
+		return err
+	}
+	defer maxRows.Close()
```

Comment:
Why does it need two queries? Is there any way to do just one?

If it's truly necessary, I think it should run them concurrently using `errgroup` to reduce latency.

Reply:
it needs 2 queries see the [description](https://github.com/rilldata/rill/pull/2584#issue-1747512744). Changed to run in parallel.